zircon.core/common/src/main/kotlin/org/hexworks/zircon/internal/component/impl/DefaultHorizontalNumberInput.kt

```diff
@@ -0,0 +1,30 @@
+package org.hexworks.zircon.internal.component.impl
+
+import org.hexworks.zircon.api.component.NumberInput
+import org.hexworks.zircon.api.component.data.ComponentMetadata
+import org.hexworks.zircon.api.component.renderer.ComponentRenderingStrategy
+import org.hexworks.zircon.internal.Zircon
+import org.hexworks.zircon.internal.event.ZirconEvent
+import org.hexworks.zircon.internal.event.ZirconScope
+import kotlin.math.min
+
+class DefaultHorizontalNumberInput(
+        initialValue: Int,
+        maxValue: Int,
+        componentMetadata: ComponentMetadata,
+        renderingStrategy: ComponentRenderingStrategy<NumberInput>)
+    : DefaultNumberInput(
+        initialValue, maxValue, componentMetadata, renderingStrategy
+) {
+    override var maxNumberLength = min(Int.MAX_VALUE.toString().length, size.width)
+
+    override fun refreshCursor() {
+        var pos = textBuffer.cursor.position
```

Comment:
You can simplify this to 
```kotlin
val pos = textBuffer.cursor.position
  .withX(min(pos.x, contentSize.width))
  .withY(0)
```

Reply:
it won't let me reference pos in (min(pos.x, ...) if I don't do it that way :(

Reply:
Would not it be better if I simply used a Position.create() call here ?

Reply:
Oh I missed that part, sorry. Just keep it as it is now, don't worry.